### ISSUE TITLE
Fetch: Exclude certain properties from API response by custom rules

### ISSUE THREAD

**Issue by vincentrohde** [agentejo/cockpit#1053]
Is there a way to exclude/ include certain things from the API response by declaring custom rules. 

For example I would like to exclude all additional information from the CMS. These properties usually start with an _. Is there a way to exclude all properties **starting with an _** or the **field** property?

<img width="306" alt="underscore" src="https://user-images.githubusercontent.com/25182140/54132888-54401a80-4415-11e9-82d1-7c3864b27ce9.png">

<img width="489" alt="field" src="https://user-images.githubusercontent.com/25182140/54132897-5904ce80-4415-11e9-9236-d561ba91494f.png">

In the documentation, I can only find sparsely information about these two properties(filter and fields), in the JSON.stringify argument:

<img width="872" alt="entries" src="https://user-images.githubusercontent.com/25182140/54132367-653c5c00-4414-11e9-9a89-f1ce4706d9e8.png">

**Comment by pauloamgomes** [comment]
You can do that in a event like:

```
$app->on('collections.find.after', function ($name, &$entries) use ($app) {
  $ufields = ['_o', '_pid', '_by', '_mby', '_modified', '_created'];
  foreach ($entries as $idx => $entry) {
    foreach ($ufields as $ufield) {
      if (array_key_exists($ufield, $entry)) {
        unset($entries[$idx][$ufield]);
      }
    }
  }
});

**Comment by aheinze** [comment]
@pauloamgomes but only if it is an api request, otherwise this can break the backend ;-)

**Comment by pauloamgomes** [comment]
> @pauloamgomes but only if it is an api request, otherwise this can break the backend ;-)

Yes, for sure, but is the only situation I think that (exclude some fields) can be required.

**Comment by aheinze** [comment]
another option is 

```
fields: {'_o':-1, '_pid':-1, '_by':-1, '_mby':-1, '_modified':-1, '_created':-1}
```

or adjust the fields property in the read rules (in the collection settings)

**Comment by vincentrohde** [comment]
> another option is
> 
> ```
> fields: {'_o':0, '_pid':0, '_by':0, '_mby':0, '_modified':0, '_created':0}
> ```
> or adjust the fields property in the read rules (in the collection settings)

Where do you implement this? If I put this into my fetch argument, the fields are not excluded from the API response. Am I missing something here?

**Comment by G-Rath** [comment]
I've run into this myself - `collections.find.before` gives you `&$options`, & `collections.find.after` gives you `&$entries`, but not the "other", which greatly limits what you can do with them :/

I feel like something is missing from somewhere - is it maybe possible to get the options from `$app` somehow?

**Comment by G-Rath** [comment]
Heres the first iteration of a plugin I made to solve this problem: https://gist.github.com/G-Rath/934fabe5ed91fefe61d6f3e656527cd5

It's rough, but the basic idea is there - hopefully it'll be helpful to someone :)

Credit appreciated if used, as I'm interested in seeing what it actually gets used for 😉 

When/if I get some time, I plan to try and make it into a proper plugin that hopefully has a frontend component.

**Comment by teamgroove** [comment]
Also, when you send:
{ "simple": true }
There will be no fields-schema on top of the response.
